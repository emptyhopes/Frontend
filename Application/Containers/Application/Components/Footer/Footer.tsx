import styled from "styled-components";

import { Container } from "@/Application/Containers/Application/Components/Container/Container";

const StyledFooter = styled.footer`
  height: 120px;
  background-color: #fffadd;
`;

const Footer = () => {
  return (
    <>
      <StyledFooter>
        <Container></Container>
      </StyledFooter>
    </>
  );
};

export { Footer };
