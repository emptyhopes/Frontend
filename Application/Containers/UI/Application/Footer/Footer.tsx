import styled from "styled-components";

import { Container } from "@/Application/Containers/UI/Application/Container/Container";
import { Content } from "@/Application/Containers/UI/Application/Content/Content";

const StyledFooter = styled.footer`
  height: 120px;
  flex: 0 0 auto;
  background-color: var(--footer-background-color);
`;

const Footer: React.FunctionComponent = (): React.ReactElement => {
  return (
    <>
      <StyledFooter>
        <Container>
          <Content flex VerticalCenter HorizontalCenter $width $height>
            <h1>Footer</h1>
          </Content>
        </Container>
      </StyledFooter>
    </>
  );
};

export { Footer };
