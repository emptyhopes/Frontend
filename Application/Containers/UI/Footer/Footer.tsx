import styled from "styled-components";

import { FunctionComponent } from "react";

import { Container } from "@/Application/Containers/UI/Container/Container";

const StyledFooter = styled.footer`
  height: 120px;
  background-color: #fffadd;
`;

const Footer: FunctionComponent = () => {
  return (
    <>
      <StyledFooter>
        <Container></Container>
      </StyledFooter>
    </>
  );
};

export { Footer };
