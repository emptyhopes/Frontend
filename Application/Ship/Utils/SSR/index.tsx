import * as React from "react";
import * as ReactDOM from "react-dom";

import { Application } from "@/Application/Containers/Application/Application";

const Structures = <Application />;

const element = document.getElementById("root");

ReactDOM.hydrate(Structures, element);
