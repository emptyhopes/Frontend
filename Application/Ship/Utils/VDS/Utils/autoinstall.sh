#! /bin/bash

# Update and upgrade system
sudo apt-get update --yes && sudo apt-get upgrade --yes

# Remove not use packages
sudo apt-get autoremove --yes

# Install packages
sudo apt-get install git nginx certbot python3-certbot-nginx --yes

# SWAP

sudo fallocate -l 2048M /swapfile
sudo chmod 600 /swapfile
sudo mkswap /swapfile
sudo swapon /swapfile

sudo cp /etc/fstab /etc/fstab.back

echo "/swapfile none swap sw 0 0" | sudo tee -a /etc/fstab

# NVM ( node@latest and npm@latest )

curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.39.2/install.sh | bash

source ~/.nvm/nvm.sh

nvm install node

npm install --global npm@latest

# Install letsencrypt SSL certificate

sudo certbot
