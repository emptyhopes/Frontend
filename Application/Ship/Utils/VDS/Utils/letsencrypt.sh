#! /bin/bash

hostname="emptyhopes.ru"

sudo certbot certonly --nginx -d $hostname