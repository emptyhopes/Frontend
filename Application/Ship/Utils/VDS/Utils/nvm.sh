#! /bin/bash

temporary=$(mktemp --directory)

curl --silent --location https://raw.githubusercontent.com/nvm-sh/nvm/v0.39.2/install.sh >"$temporary/install.sh"

bash "$temporary/install.sh"

source ~/.nvm/nvm.sh

nvm install node
