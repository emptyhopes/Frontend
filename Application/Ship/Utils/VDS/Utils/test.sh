#! /bin/bash

hostname="gartmann-art.ru"

curl --location -I http://$hostname.ru
curl --location -I http://en.$hostname.ru
curl --location -I http://ru.$hostname.ru

curl --location -I http://www.$hostname.ru
curl --location -I http://www.en.$hostname.ru
curl --location -I http://www.ru.$hostname.ru

curl --location -I https://$hostname.ru
curl --location -I https://en.$hostname.ru
curl --location -I https://ru.$hostname.ru

curl --location -I https://www.$hostname.ru
curl --location -I https://www.en.$hostname.ru
curl --location -I https://www.ru.$hostname.ru