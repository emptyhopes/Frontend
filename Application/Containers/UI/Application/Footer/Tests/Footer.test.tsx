import { RenderTests } from "@/Application/Ship/Tests/Utils/RenderTests";

import { Footer } from "@/Application/Containers/UI/Application/Footer/Footer";

test("Footer", () => {
  const { container } = RenderTests(<Footer />);

  expect(container).toBeInTheDocument();
  expect(container).toMatchSnapshot();
});
