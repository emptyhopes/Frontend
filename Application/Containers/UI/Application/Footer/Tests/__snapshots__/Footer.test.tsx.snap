// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer 1`] = `
.c1 {
  width: 100%;
  max-width: 1240px;
  height: 100%;
  padding: 0 10px;
  margin: 0 auto;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  height: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c0 {
  height: 120px;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  background-color: var(--footer-background-color);
}

<div>
  <footer
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <h1>
          Footer
        </h1>
      </div>
    </div>
  </footer>
</div>
`;
